javatests/arcs/android/storage/handle/AndroidHandleManagerTest.kt

```diff
@@ -0,0 +1,129 @@
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.work.testing.WorkManagerTestInitHelper
+import arcs.android.storage.ParcelableStoreOptions
+import arcs.android.storage.handle.TestActivity
+import arcs.android.storage.service.IStorageService
+import arcs.core.data.FieldType
+import arcs.core.data.RawEntity
+import arcs.core.data.Schema
+import arcs.core.data.SchemaDescription
+import arcs.core.data.SchemaFields
+import arcs.core.data.SchemaName
+import arcs.core.data.util.toReferencable
+import arcs.core.storage.handle.HandleManager
+import arcs.sdk.android.storage.service.DefaultConnectionFactory
+import arcs.sdk.android.storage.service.StorageService
+import arcs.sdk.android.storage.service.StorageServiceBindingDelegate
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.Robolectric
+import org.robolectric.android.controller.ServiceController
+
+
+@Suppress("EXPERIMENTAL_API_USAGE")
+@RunWith(AndroidJUnit4::class)
+class AndroidHandleManagerTest {
+    private lateinit var app: Application
+
+    inner class TestBindingDelegate : StorageServiceBindingDelegate {
+        var sc: ServiceController<StorageService>? = null
+        override fun bindStorageService(conn: ServiceConnection, flags: Int, options: ParcelableStoreOptions): Boolean {
+            val intent = StorageService.createBindIntent(
+                app,
+                options
+            )
+            sc = Robolectric.buildService(StorageService::class.java, intent)
+                .create()
+                .bind()
+                .also {
+                    val binder = it.get().onBind(intent)
+                    conn.onServiceConnected(null, binder)
+                }
+            return true
+        }
+
+        override fun unbindStorageService(conn: ServiceConnection) {
+            sc?.destroy()
+        }
+    }
+
+    val entity1 = RawEntity("empty", singletons=mapOf(
+        "name" to "Jason".toReferencable(),
+        "age" to 21.toReferencable(),
+        "is_cool" to false.toReferencable()
+    ), collections=emptyMap())
+
+    private val schema = Schema(
+        listOf(SchemaName("Person")),
+        SchemaFields(
+            singletons = mapOf(
+                "name" to FieldType.Text,
+                "age" to FieldType.Number,
+                "is_cool" to FieldType.Boolean
+            ),
+            collections = emptyMap()
+        ),
+        SchemaDescription(),
+        "1234acf"
+    )
+
+    @Before
+    fun setUp() {
+        app = ApplicationProvider.getApplicationContext()
+        app.setTheme(R.style.Theme_AppCompat);
+
+        // Initialize WorkManager for instrumentation tests.
+        WorkManagerTestInitHelper.initializeTestWorkManager(app)
+    }
+
+    fun handleFactoryTest(block: suspend (HandleManager) -> Unit) {
```

Comment:
Private? And maybe move it to the bottom of the class, so the `@Test`s are front-and-center.